src/races/races.service.ts

```diff
@@ -294,4 +296,534 @@ export class RacesService {
 
     return [...uniqueRaces.values()]
   }
+
+  private getRaceLevel(level: string) {
+    // this helper just simplifies level to city/state/county/federal.
+    // it is used in p2v.
+    // from this larger list:
+    // "level"
+    // "city"
+    // "county"
+    // "federal"
+    // "local"
+    // "regional"
+    // "state"
+    // "town"
+    // "township"
+    // "village"
+
+    try {
+      level = level.toLowerCase()
+      if (
+        level &&
+        level !== 'federal' &&
+        level !== 'state' &&
+        level !== 'county' &&
+        level !== 'city'
+      ) {
+        level = 'city'
+      }
+      return level
+    } catch (e) {
+      this.logger.log('error at get-race-level', e)
+      return level
+    }
+  }
+
+  private async resolveMtfcc(geoId: string, mtfcc: string) {
+    let geoData: any
+    // geoId is a string that an start with 0, so we need remove that 0
+    if (geoId) {
+      geoId = parseInt(geoId, 10).toString()
+    }
+    if (mtfcc && geoId) {
+      const census = await this.prisma.censusEntity.findFirst({
+        where: {
+          geoId,
+          mtfcc,
+        },
+      })
+      if (census) {
+        geoData = {
+          name: census.name,
+          type: census.mtfccType,
+        }
+
+        // todo: this can be improved for county recognition
+        // and other types of entities (school board, etc)
+
+        // if (census.mtfccType !== 'State or Equivalent Feature') {
+        //   geoData.city = census.name;
+        // }
+        if (census.mtfccType === 'Incorporated Place') {
+          geoData.city = census.name
+        } else if (census.mtfccType === 'County or Equivalent Feature') {
+          // todo: strip County from name.
+          geoData.county = census.name
+        } else if (census.mtfccType === 'State or Equivalent Feature') {
+          geoData.state = census.name
+        } else if (census.mtfccType === 'County Subdivision') {
+          if (census.name.toLowerCase().includes('township')) {
+            geoData.township = census.name
+          } else if (census.name.toLowerCase().includes('town')) {
+            geoData.town = census.name
+          } else if (census.name.toLowerCase().includes('city')) {
+            geoData.city = census.name
+          } else if (census.name.toLowerCase().includes('village')) {
+            geoData.village = census.name
+          } else if (census.name.toLowerCase().includes('borough')) {
+            geoData.borough = census.name
+          }
```

Comment:
All of these string literals need to be placed into `enum`s and referenced as such